java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java

```diff
@@ -571,8 +571,10 @@ public int getByteCapacity() {
 
   @Override
   public int getCurrentSizeInBytes() {
-    /* TODO */
-    return 0;
+    if (valueCount == 0) {
+      return 0;
+    }
+    return offsetBuffer.getInt(valueCount * OFFSET_WIDTH);
```

Comment:
I think you should probably include valueBuffer somehow as well?

Reply:
Thanks for your comments, I will add a test later. 
And what do you mean "includeBuffer"? It seems getCurrentSizeInBytes means the valueBuffer size in Drill.

Reply:
I was thinking it should be offset buffer + size of valueBuffer (I think the last value in offsetBuffer) as that would be the total size but I guess it isn't well defined (no comments on intent on the javadoc)

Reply:
Not sure, you mean the implementation like this？
`final int offsetBufferSize = (valueCount + 1) * OFFSET_WIDTH;`
 `final int dataBufferSize = offsetBuffer.getInt(valueCount * OFFSET_WIDTH);`
 ` return offsetBufferSize + dataBufferSize;`

if so, why not include validityBuffer, then this method is duplicated with getBufferSizeFor.
And this method in drill : https://github.com/apache/drill/blob/master/exec/vector/src/main/codegen/templates/VariableLengthVectors.java#L125

Reply:
Thank you for the pointer.  It seems reasonable to mimic what they are doing?  Could you document on the interface what will be returned?

Reply:
Sure~

Reply:
I am not sure following the Drill precedent is right, can we create a new method that explicitly calls out that this sizeOfValueBuffer alone?

Reply:
@praveenbingo Thanks for your comments, do you mean we just remove getCurrentSizeInBytes in VariableWidthVector and create a new method sizeOfValueBuffer instead? Not sure about that, what do you think? @emkornfield If you both think it is ok, I can update this PR, thanks!